[ISSUE] Websockets Not Working

Hi, 

### Websockets not working / syncing between Chrome Client and Web Page
I had websockets working previously, but I am not getting this error:

```WebSocket Protocol Error: Unable to parse WebSocket key.```

The error appears when accessing both the internal IP of ``192.168.0.11:3012`` and the external server through the internet ``https://vault.example.in/notifications/hub``

My Docker Compose is as follows:

````  bitwarden:
    container_name: bitwarden
    image: bitwardenrs/server
    restart: unless-stopped
    volumes:
      - "/etc/localtime:/etc/localtime:ro"
      - "/opt/Bitwarden:/data"
    environment:
      LOG_FILE: /data/bitwarden.log
      WEBSOCKET_ENABLED: "true"
      INVITATIONS_ALLOWED: "false"
      SIGNUPS_ALLOWED: "false"
      SHOW_PASSWORD_HINT: "false"
      DOMAIN: https://vault.example.in
      ADMIN_TOKEN: xxxxx
    ports:
      - 8008:80
      - 3012:3012
````

My NGINX Reverse proxy is as follows:
````#Bitwarden Block
#Add entry in Cloudflare DNS ("CNAME vault example.in") to enable
server {
  #BlockedAgent
  if ($blockedagent) {
        return 403;
  }
  
  #Bad Bots Filtering
  if ($limit_bots = 1) {
            return 403;
  }
  
  #listen [::]:443 ssl http2;
  listen 443 ssl http2;
  server_name vault.example.in;
  include /config/nginx/conf.d/*.conf;
  add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
  
    #client_max_body_size 128M;
      
    #SSL Configuration
    include /config/nginx/ssl.conf;

    location / {
    proxy_pass http://192.168.0.11:8008;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    }
  
    location /notifications/hub {
    proxy_pass http://192.168.0.11:3012;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
    }
  
    location /notifications/hub/negotiate {
    proxy_pass http://192.168.0.11:8008;
    }

    location /admin {
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;

    proxy_pass http://192.168.0.11:8008;
  } 
}
````

### Your environment
<!-- The version number, obtained from the logs or the admin page -->
* Bitwarden_rs version: Version 1.14.1
* Install method: Docker (Docker Compose)
* Clients used: Web Page ``https://vault.example.in`` and Chrome Addon Client
* Reverse proxy and version: Nginx 1.16.1

### Expected behaviour
Websocket Sync to happen instantly upon changes in either client

### Actual behaviour
Websocket Sync is not happening upon changes in either client

### Relevant logs
````[2020-03-26 17:21:14][ws::io][INFO] Accepted a new tcp connection from 172.18.0.1:39958.
[2020-03-26 17:21:14][ws::handler][ERROR] WS Error <Io(Kind(InvalidData))>: Missing id or access token
[2020-03-26 17:21:19][ws][INFO] Listening for new connections on 0.0.0.0:3012.
[2020-03-26 17:21:19][start][INFO] Rocket has launched from http://0.0.0.0:80
[2020-03-26 17:21:22][ws::io][INFO] Accepted a new tcp connection from 172.18.0.1:39972.
[2020-03-26 17:21:22][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:21:23][ws::io][INFO] Accepted a new tcp connection from 172.18.0.1:39978.
[2020-03-26 17:21:23][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:21:23][ws::io][INFO] Accepted a new tcp connection from 172.18.0.1:39982.
[2020-03-26 17:21:23][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:21:24][ws::io][INFO] Accepted a new tcp connection from 172.18.0.1:39986.
[2020-03-26 17:21:24][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:21:24][ws::io][INFO] Accepted a new tcp connection from 172.18.0.1:39990.
[2020-03-26 17:21:24][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:21:25][ws::io][INFO] Accepted a new tcp connection from 172.18.0.1:39994.
[2020-03-26 17:21:25][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:21:31][ws::io][INFO] Accepted a new tcp connection from 172.18.0.1:40040.
[2020-03-26 17:21:31][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:21:31][ws::io][INFO] Accepted a new tcp connection from 172.18.0.1:40044.
[2020-03-26 17:21:31][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:21:31][ws::io][INFO] Accepted a new tcp connection from 172.18.0.1:40048.
[2020-03-26 17:21:31][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:22:04][ws::io][INFO] Accepted a new tcp connection from 192.168.0.23:53163.
[2020-03-26 17:22:04][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:22:04][ws::io][INFO] Accepted a new tcp connection from 192.168.0.23:53164.
[2020-03-26 17:22:04][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:22:04][ws::io][INFO] Accepted a new tcp connection from 192.168.0.23:53165.
[2020-03-26 17:22:04][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:22:04][ws::io][INFO] Accepted a new tcp connection from 192.168.0.23:53166.
[2020-03-26 17:22:04][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:22:05][ws::io][INFO] Accepted a new tcp connection from 192.168.0.23:53167.
[2020-03-26 17:22:05][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:22:05][ws::io][INFO] Accepted a new tcp connection from 192.168.0.23:53168.
[2020-03-26 17:22:05][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:22:41][ws::io][INFO] Accepted a new tcp connection from 192.168.0.22:42262.
[2020-03-26 17:22:41][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
[2020-03-26 17:22:41][ws::io][INFO] Accepted a new tcp connection from 192.168.0.22:42264.
[2020-03-26 17:22:41][ws::handler][ERROR] WS Error <Protocol>: Unable to parse WebSocket key.
````

--- a/src/api/notifications.rs
+++ b/src/api/notifications.rs
@@ -152,6 +152,9 @@ impl WSHandler {
 impl Handler for WSHandler {
     fn on_open(&mut self, hs: Handshake) -> ws::Result<()> {
         // Path == "/notifications/hub?id=<id>==&access_token=<access_token>"
+        //
+        // We don't use `id`, and as of around 2020-03-25, the official clients
+        // no longer seem to pass `id` (only `access_token`).
         let path = hs.request.resource();
 
         let (_id, access_token) = match path.split('?').nth(1) {
@@ -170,10 +173,11 @@ impl Handler for WSHandler {
 
                 match (id, access_token) {
                     (Some(a), Some(b)) => (a, b),
-                    _ => return self.err("Missing id or access token"),
+                    (None, Some(b)) => ("", b), // Ignore missing `id`.
+                    _ => return self.err("Missing access token"),
                 }
             }
-            None => return self.err("Missing query path"),
+            None => return self.err("Missing query parameters"),
         };
 
         // Validate the user

